### PR TITLE
Heroku deploy support

### PR DESCRIPTION
**Pull request by bmann**
Hey there! 

I saw a previous attempt to get this working, and you already have a Procfile in the root that Heroku needs.

I have tested this and done a couple of installs. Getting Google or Slack auth login support working is the hardest part, with Slack being easier. I still had a bit of difficulty -- probably walking people through what they need to do would be helpful.

Also, SLACK VERIFICATION is, I think deprecated, and might not be needed anymore -- both in your `env.sample` and in `app.json`.

You can do a test yourself by following this link, which will deploy from the branch that I am PR'ing here: https://heroku.com/deploy?template=https://github.com/bmann/outline/tree/heroku-deploy-support

### CODE REVIEW

**Comment by tommoor** [comment]
👋Hey, thanks for putting this together.

`SLACK_VERIFICATION_TOKEN` is required if you want to respond to hooks from Slack, but it's not required for signin.

File: app.json [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      "AWS_S3_UPLOAD_BUCKET_NAME": {
+        "description": "yourbucket.example.com",
+        "required": false
+      },
+      "AWS_S3_UPLOAD_BUCKET_URL": {
+        "description": "Live web link to your bucket. For CNAMEs, https://yourbucket.example.com",
+        "required": false
+      },
+      "AWS_S3_UPLOAD_MAX_SIZE": {
+        "description": "",
```
Comment (tommoor) [inline app.json]:
```suggestion
        "description": "Maximum file upload size in bytes",
```

File: app.json [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      "SMTP_REPLY_EMAIL": {
+        "description": "wikireply@example.com (optional)",
+        "required": false
+      },
+      "GOOGLE_ANALYTICS_ID": {
+        "description": "UA-xxxx (optional)",
+        "required": false
+      },
+      "BUGSNAG_KEY": {
+        "description": "(optional)",
```
Comment (tommoor) [inline app.json]:
```suggestion
        "description": "An API key for bugsnag if you wish to collect error reporting (optional)",
```

File: app.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        "required": false
+      },
+      "GITHUB_ACCESS_TOKEN": {
+        "description": "(optional)",
```
Comment (tommoor) [inline app.json]:
```suggestion
        "description": "An API token for GitHub, optional for self hosted (optional)",
```

File: app.json [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        "required": true
+      },
+      "ENABLE_UPDATES": {
+        "value": "true",
+        "required": true
+      },
+      "SUBDOMAINS_ENABLED": {
+        "value": "false",
+        "required": true,
+        "description": "Running different subdomains. Not needed in single org mode"
```
Comment (tommoor) [inline app.json]:
```suggestion
        "description": "Allows each team to have a different subdomain. Not recommend when self hosting"
```

File: app.json [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      "postdeploy": "yarn sequelize db:migrate"
+    },
+    "env": {
+      "SECRET_KEY": {
+        "description": "A secret key",
+        "generator": "secret",
+        "required": true
+      },
+      "DEPLOYMENT": {
+        "value": "self",
```
Comment (tommoor) [inline app.json]:
```suggestion
		"description": "Should be 'self' for self hosted installations, turns off things like pricing pages"
        "value": "self",
```

**Comment by bmann** [comment]
I haven't used that Github suggested commits before. Super slick! Thanks @tommoor for making that easy! Should be ready to go.

**Comment by tommoor** [comment]
Great, thanks!

**Comment by satyadeepk** [comment]
Awesome! You can also probably add a "Deploy to Heroku" button in the Readme.

**Comment by bmann** [comment]
Yeah. I think especially with your hosting model, sticking it in docs or somewhere else makes more sense than the README.

Very few people are going to be like me and yak shave their way through Slack and/or Google OAuth ;)

Thanks for accepting!

**Comment by tommoor** [comment]
Full transparency – I definitely questioned whether to accept the PR or not 😉. Like you say the long-term business model here is to make hosting and maintaining Outline easy but I believe what you said is correct, there is still enough burden and technical know-how required to maintain your own copy.

**Pull request merged**